### PR TITLE
Autorization rules

### PR DESCRIPTION
**Pull request by HectorTorresE**
### In this PR

- Install CanCanCan
- Add a role column to the user's table.
- Add delete a post function
- Add delete comment function

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -27,6 +27,11 @@ def create
     end
   end
 
+  def destroy
+    post = Post.find(params[:id])
+    redirect_to user_path(post.author_id) if Post.destroy(params[:id])
```
Comment (arslanbisharat) [inline app/controllers/posts_controller.rb:33]: - [ ] As per the guidelines provided by Micoverse please make sure you've used **CanCanCan** for Post authorization in `Post Controller`. 

```ruby
class PostsController < ApplicationController
    load_and_authorize_resource

    ...
  end
```

**Review by arslanbisharat — APPROVED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- The Pull Request has a proper title and description. ✔️
- You have added a descriptive Readme file. Good Job. ✔️
- There is no linter error is present. ✔️
- Project has met all the technical aspects you needed for this project. Keep rocking! :x:

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] Please make sure you right descriptive comments. Commenting involves placing Human Readable Descriptions inside of computer programs detailing what the Code is doing. Proper use of commenting can make code maintenance much easier, as well as helping make finding bugs faster. Further, commenting is very important when writing functions that other people will use. :x:


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (Santiago220991) [reply app/controllers/posts_controller.rb:33]: Hi, @arslanbisharat. I'm going to add the load_and_authorize_resource method in the Post Controller. Thanks. :+1:

Comment (Whoistolu) [reply app/controllers/posts_controller.rb:33]: Well resolved 👍🏽

**Review by Whoistolu — APPROVED** [review]
### Hi Team,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Santiago220991** [comment]
Thanks. :+1:

**Pull request merged**